src/test/java/ru/yandex/practicum/filmorate/storage/RecommendationTest.java

```diff
@@ -0,0 +1,89 @@
+package ru.yandex.practicum.filmorate.storage;
+
+import lombok.RequiredArgsConstructor;
+import org.junit.jupiter.api.Test;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.boot.test.autoconfigure.jdbc.AutoConfigureTestDatabase;
+import org.springframework.boot.test.autoconfigure.web.servlet.AutoConfigureMockMvc;
+import org.springframework.boot.test.context.SpringBootTest;
+import org.springframework.http.MediaType;
+import org.springframework.test.web.servlet.MockMvc;
+import org.springframework.test.web.servlet.request.MockMvcRequestBuilders;
+import ru.yandex.practicum.filmorate.controllers.FilmController;
+import ru.yandex.practicum.filmorate.controllers.UserController;
+import ru.yandex.practicum.filmorate.storage.film.FilmDbStorage;
+
+import static org.springframework.test.web.servlet.result.MockMvcResultHandlers.print;
+import static org.springframework.test.web.servlet.result.MockMvcResultMatchers.content;
+import static org.springframework.test.web.servlet.result.MockMvcResultMatchers.status;
+
+@SpringBootTest
+@AutoConfigureMockMvc
+@AutoConfigureTestDatabase
+@RequiredArgsConstructor(onConstructor_ = @Autowired)
+public class RecommendationTest {
+    @Autowired
+    UserController userController;
+    @Autowired
+    FilmController filmController;
+    @Autowired
+    FilmDbStorage filmDbStorage;
+
+    @Autowired
+    private MockMvc mockMvc;
+
+    //пользователю 3 рекомендуются фильмы 4, 6, 2 тк у них пересечение с пользователем 2 по оценке фильма 3
+    @Test
+    void getRecommendationUser3() throws Exception {
+        mockMvc.perform(MockMvcRequestBuilders.get("/users/3/recommendations"))
+                .andExpect(status().isOk())
+                .andDo(print())
+                .andExpect(content().contentTypeCompatibleWith(MediaType.APPLICATION_JSON))
+                .andExpect(content()
+                        .json("[{\"id\":4,\"name\":\"Фильм4\",\"description\":\"какое-то описание\"," +
```

Comment:
[Recommendation] Avoid comparing json string, it's hard to read. Parse JSON to JSON node, for instance and validate fields.